app/models/bulk_upload.rb

```diff
@@ -11,11 +12,56 @@ class BulkUpload < ApplicationRecord
 
   after_initialize :generate_identifier, unless: :identifier
 
+  scope :search_by_filename, ->(filename) { where("filename ILIKE ?", "%#{filename}%") }
+  scope :search_by_user_name, ->(name) { where(user_id: User.where("name ILIKE ?", "%#{name}%").select(:id)) }
+  scope :search_by_user_email, ->(email) { where(user_id: User.where("email ILIKE ?", "%#{email}%").select(:id)) }
+  scope :search_by_organisation_name, ->(name) { where(organisation_id: Organisation.where("name ILIKE ?", "%#{name}%").select(:id)) }
```

Comment:
I think joins(:organisation)/(:user) might be more performant than a subquery

Reply:
If I remember right, joining organisation brought up errors wanting to join organisation also on the other search scopes. This was a way to avoid doing that.